pkg/m3admin/namespace.go

```diff
@@ -0,0 +1,115 @@
+package m3admin
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/gogo/protobuf/jsonpb"
+	namespace "github.com/m3db/m3/src/dbnode/generated/proto/namespace"
+	"github.com/m3db/m3/src/query/generated/proto/admin"
+	myspec "github.com/m3db/m3db-operator/pkg/apis/m3dboperator/v1"
+)
+
+const (
+	// TODO(PS) add to spec for crd?
+	// defaults for namespace creation
```

Comment:
Could you use the simplified endpoints exposed for this --
```
curl -X POST http://localhost:7201/api/v1/database/create -d '{
  "type": "local",
  "namespaceName": "default",
  "retentionTime": "48h"
}'
```

 The concern is to keep these defaults in sync across multiple repos (e.g. say we add a parameter to the namespace options) is error-prone and unnecessary.

Reply:
So the thinking behind the `database/create/` API was to provide a single endpoint with sensible defaults to make it easier for users to create a placement/namespace without thinking about the details. It basically wraps the placement/namespace APIs which have the more granular CRUD operations with some heuristics for sizing. 

How about migrating to use the create for the creation, and using the placement/namespace APIs for the more granular ops?